infrastructure/terraform/per_workspace/modules/etl/sds/main.tf

```diff
@@ -432,3 +440,36 @@ module "bulk_trigger_notification" {
   filter_prefix = "${local.bulk_trigger_prefix}/"
   filter_suffix = ".ldif"
 }
+
+module "proxy_lambda_executor" {
+  source = "./proxy-executor/"
+
+  executor_name         = "input"
+  etl_name              = local.etl_name
+  workspace_prefix      = var.workspace_prefix
+  python_version        = var.python_version
+  event_layer_arn       = var.event_layer_arn
+  third_party_layer_arn = var.third_party_sds_update_layer_arn
+  sds_layer_arn         = var.sds_layer_arn
+  etl_bucket_arn        = module.bucket.s3_bucket_arn
+  etl_layer_arn         = module.etl_layer.lambda_layer_arn
+  notify_lambda_arn     = module.notify.arn
+  state_machine_arn     = aws_sfn_state_machine.state_machine.arn
+  table_arn             = var.table_arn
+  allowed_triggers      = {}
+  environment_variables = {
+    # prepend '/opt/python' (the root path of the layer) to LD_LIBRARY_PATH so that
+    # all compiled dependencies can find each other. Note: this is a hack - and
+    # may result in version mismatches between system libs on the lambda. The stable
+    # alternative is to run or deploy the service from a container.
+    LD_LIBRARY_PATH         = "/opt/python:/var/lang/lib:/lib64:/usr/lib64:/var/runtime:/var/runtime/lib:/var/task:/var/task/lib:/opt/lib"
+    TRUSTSTORE_BUCKET       = var.truststore_bucket.id
+    CPM_FQDN                = "cpm.thirdparty.nhs.uk"
+    LDAP_HOST               = data.aws_secretsmanager_secret_version.ldap_host.secret_string
+    LDAP_CHANGELOG_USER     = data.aws_secretsmanager_secret_version.ldap_changelog_user.secret_string
+    LDAP_CHANGELOG_PASSWORD = data.aws_secretsmanager_secret_version.ldap_changelog_password.secret_string
```

Comment:
i think most of these env vars aren't needed for the state lock lambda?

Reply:
Ah yes, lots are unnecessary. Also removed the step function info/permissions from the bulk/update triggers as only the input lambda needs this now.